apps/tangle-dapp/constants/evm.ts

```diff
@@ -8,76 +8,118 @@ import {
+    account: ensureHex(nominatorAddress),
     transport: custom(window.ethereum),
   });
 
-  const amountToBond = parseEther(amount.toString());
+export const bondTokens = async (
+  nominatorAddress: string,
+  numberOfTokens: number,
+  paymentDestination: string
+): Promise<`0x${string}`> => {
+  const value = parseEther(numberOfTokens.toString());
+  const payee =
+    paymentDestination === PaymentDestination.AUTO_COMPOUND
+      ? PAYEE_STAKED
+      : PAYEE_STASH;
 
-  const payee = payeeStaked;
+  try {
+    const { request } = await evmPublicClient.simulateContract({
+      address: StakinInterfaceAddress,
+      abi: StakinInterfaceABI,
+      functionName: 'bond',
+      args: [value, payee],
+      account: ensureHex(nominatorAddress),
+    });
 
-  const targets = selectedValidators.map((addr) => {
-    return u8aToHex(decodeAddress(addr));
-  });
+    const evmWalletClient = createEvmWalletClient(nominatorAddress);
+
+    const txHash = await evmWalletClient.writeContract(request);
+
+    return txHash;
+  } catch (e: any) {
+    throw new Error(e);
+  }
+};
+
+export const bondExtraTokens = async (
+  nominatorAddress: string,
+  numberOfTokens: number
+): Promise<`0x${string}`> => {
+  const value = parseEther(numberOfTokens.toString());
 
   try {
-    const bondTx = await client.writeContract({
-      address: contractAddress,
+    const { request } = await evmPublicClient.simulateContract({
+      address: StakinInterfaceAddress,
       abi: StakinInterfaceABI,
-      functionName: 'bond',
-      args: [amountToBond, payee],
+      functionName: 'bondExtra',
+      args: [value],
+      account: ensureHex(nominatorAddress),
     });
 
-    if (bondTx) {
-      console.log('bondTx', bondTx);
+    const evmWalletClient = createEvmWalletClient(nominatorAddress);
+
+    const txHash = await evmWalletClient.writeContract(request);
+
+    return txHash;
+  } catch (e: any) {
+    throw new Error(e);
+  }
+};
+
+export const nominateValidators = async (
+  nominatorAddress: string,
+  validatorAddresses: string[]
+): Promise<`0x${string}`> => {
+  const targets = validatorAddresses.map((address) => {
+    return u8aToHex(decodeAddress(address));
+  });
+
+  try {
+    const { request } = await evmPublicClient.simulateContract({
+      address: StakinInterfaceAddress,
+      abi: StakinInterfaceABI,
+      functionName: 'nominate',
+      args: [targets],
+      account: ensureHex(nominatorAddress),
+    });
 
-      const nominateTx = await client.writeContract({
-        address: contractAddress,
-        abi: StakinInterfaceABI,
-        functionName: 'nominate',
-        args: [targets],
-      });
+    const evmWalletClient = createEvmWalletClient(nominatorAddress);
 
-      console.log('nominateTx', nominateTx);
+    const txHash = await evmWalletClient.writeContract(request);
 
-      return nominateTx;
-    }
-  } catch (e) {
-    console.log(e);
+    return txHash;
+  } catch (e: any) {
+    throw new Error(e);
```

Comment:
This doesn't seem like proper error handling

Reply:
If you're just going to throw an error it's no different than removing the `try/catch` altogether

Reply:
@drewstone We catch this error to notify users in this function [here](https://github.com/webb-tools/webb-dapp/blob/71f8eab8a53b4be8d9b0964834e5d09bccf853da/apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx#L105-L165). But I'll optimize the use of try/catch.